### ISSUE TITLE
Add support for other location services than MLS

### ISSUE THREAD

**Issue by mjaakko** [mjaakko/NeoStumbler#79]
Currently data can only be sent to services which implement an API that is compatible with MLS Geosubmit. Investigate adding support for other location services, such as [OpenCellID](https://www.opencellid.org/) for cell towers

The user would be able to configure the type of endpoint and its parameters (API key etc.) in the settings

**Comment by joelkoen** [comment]
Would it be worth maintaining some kind of list of MLS compatible servers?

**Comment by lucasmz-dev** [comment]
microG just shared the existence of positon.xyz. That's really the only one I know though.

**Comment by joelkoen** [comment]
I'm hosting [beaconDB](https://beacondb.net) but it's experimental and there's barely any coverage.

**Comment by lucasmz-dev** [comment]
so far I haven't seen any that does IPv6 :/
MLS was the same too, IPv4 only

**Comment by dimaguy** [comment]
Given more than a single endpoint (like positon and beacondb) is appearing to try and replace MLS, it'd be nice if NeoStumbler allowed to input more URLs to send reports to

**Comment by lucasmz-dev** [comment]
@dimaguy atm a work around is to  change the endpoint and use the reupload data option

**Comment by dimaguy** [comment]
Sounds a bit clunky for something that's supposed to be running automatically in the background, but sure that wkrks

**Comment by joelkoen** [comment]
This is a good idea, providing data to both services doesn't hurt, and is better for the community.

**Comment by mjaakko** [comment]
I've created another issue for sending data to multiple endpoints, #124

**Comment by pataquets** [comment]
Would it be a reasonable FR to request to have a dropdown list of available/curated known services for the user to choose from? This would avoid the user having to look for and copypasting URLs.
I'd happily file an issue if it is.

**Comment by mjaakko** [comment]
> Would it be a reasonable FR to request to have a dropdown list of available/curated known services for the user to choose from?

In principle I think it's a good idea. Will have to think about the details - there should be strict criteria for which services to include, e.g. should be clear who hosts the service, have links to terms of use and privacy policy, etc.

**Comment by pataquets** [comment]
> > Would it be a reasonable FR to request to have a dropdown list of available/curated known services for the user to choose from?
> 
> In principle I think it's a good idea. Will have to think about the details - there should be strict criteria for which services to include, e.g. should be clear who hosts the service, have links to terms of use and privacy policy, etc.

Of course, it's totally fair. On second thought, I'd even go as far as first defining such inclusion criteria as the previous task, so it does not place any more burden on maintainers when the feature gets implemented eventually.

**Comment by lucasmz-dev** [comment]
@codenyte that article is full of misinformation or at best irrelevant information. There is no proof that Positon is even hosted by /e/ and when confronted about it, the response is "the information was got from a private chat". And even with that, there is also no proof that /e/ has any involvement in attacks towards them, and so far it seems like the opposite. I don't like to talk about any claims they make as it just gives them more traction and gives them what they want. I feel the need to address this though, even though a wave may come from that community.

beacondb.net is great and I personally admire its vision and try to contribute to it with what I know and can do. There is no beef between them and if there is, it came from this article. Positon just has a different way to do things (e.g. a more private way to use online location services with commercial data quality).

I suspect it comes down to the use of Positon in microG, which they have historically attacked frequently.

**Comment by FedericoCeratto** [comment]
NeoStumbler is crashing when trying to submit to http://dev.libregeolocation.org:3940/v2/geosubmit ( test instance for https://libregeolocation.org/libreloc.html ). I'm not seeing any incoming packet on the server side. Any pointer? :)

**Comment by joelkoen** [comment]
Are you using the latest version? I think older versions required HTTPS, or you might want to try without a port

**Comment by FedericoCeratto** [comment]
I am using the latest version. It seems to be working reliably now that I'm using https://dev.libregeolocation.org/ (with TLS and behind Nginx). Some of the crashes are caused by flaky network connectivity.

**Comment by mjaakko** [comment]
> Some of the crashes are caused by flaky network connectivity.

If you have any logs available from the crashes, it would be helpful for investigating. The app shouldn't crash because of network problems

**Comment by lucasmz-dev** [comment]
I've had the app crash when I tried to use positon (it isn't an open API yet so expected it wouldn't work)

**Comment by dimaguy** [comment]
> +1. And now there's nowhere to send scanned reports, although the app is being updated.

There is though? BeaconDB works fine as replacement

**Comment by lucasmz-dev** [comment]
@codenyte, yeah it does. The dump BeaconDB uses only includes cell towers. AFAIK, Mozilla never made Wi-Fi and Bluetooth beacons available in public dumps.

**Comment by guglovich** [comment]
> > +1. And now there's nowhere to send scanned reports, although the app is being updated.
> 
> There is though? BeaconDB works fine as replacement

I don't know this, there are no endpoint presets, so how do users know this without looking at the github branches.
